storage_controller/src/service.rs

```diff
@@ -6317,18 +6318,39 @@ impl Service {
         // TODO: issue split calls concurrently (this only matters once we're splitting
         // N>1 shards into M shards -- initially we're usually splitting 1 shard into N).
 
+        // HADRON: set a timeout for splitting individual shards on page servers.
+        // Currently we do not perform any retry because it's not clear if page server can handle
+        // partially split shards correctly.
+        let shard_split_timeout =
+            if let Some(env::DeploymentMode::Local) = env::get_deployment_mode() {
+                Duration::MAX
+            } else {
+                Duration::MAX
+            };
```

Comment:
This is going to bite us in the hadron env. We should make it configurable.

Reply:
Done in [20ac603](https://github.com/neondatabase/neon/pull/12552/commits/20ac603c23c281a5988d164df3961dd65a4c99ff)